apps/evm_family/evm_sign_txn.c

```diff
@@ -0,0 +1,497 @@
+  if (EVM_TRANSACTION_SIZE_CAP < txn_context->transaction_size) {
+    evm_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                   ERROR_DATA_FLOW_INVALID_DATA);
+    status = false;
+  }
+  return status;
+}
+
+static void send_response(const pb_size_t which_response) {
+  evm_result_t result = init_evm_result(EVM_RESULT_SIGN_TXN_TAG);
+  result.sign_txn.which_response = which_response;
+  evm_send_result(&result);
+}
+
+static bool handle_initiate_query(const evm_query_t *query) {
+  char wallet_name[NAME_SIZE] = "";
+  char msg[100] = "";
+
+  // TODO: Handle wallet search failures - eg: Wallet ID not found, Wallet
+  // ID found but is invalid/locked wallet
+  if (!check_which_request(query, EVM_SIGN_TXN_REQUEST_INITIATE_TAG) ||
+      !validate_request_data(&query->sign_txn) ||
+      !get_wallet_name_by_id(query->sign_txn.initiate.wallet_id,
+                             (uint8_t *)wallet_name)) {
+    return false;
+  }
+
+  // TODO: need to show token symbol for token transactions
+  snprintf(msg,
+           sizeof(msg),
+           UI_TEXT_SEND_PROMPT,
+           g_evm_app->lunit_name,
+           g_evm_app->name,
+           wallet_name);
+  // Take user consent to sign the transaction for the wallet
+  if (!core_confirmation(msg, evm_send_error)) {
+    return false;
+  }
+
+  set_app_flow_status(EVM_SIGN_TXN_STATUS_CONFIRM);
+  memcpy(&txn_context->init_info,
+         &query->sign_txn.initiate,
+         sizeof(evm_sign_txn_initiate_request_t));
+  send_response(EVM_SIGN_TXN_RESPONSE_CONFIRMATION_TAG);
+  // show processing screen for a minimum duration (additional time will add due
+  // to actual processing)
+  delay_scr_init(ui_text_processing, DELAY_SHORT);
+  return true;
+}
+
+static bool fetch_valid_transaction(evm_query_t *query) {
+  bool status = false;
+
+  // TODO: stitch transactions split over multiple chunks
+  if (!evm_get_query(query, EVM_QUERY_SIGN_TXN_TAG) &&
+      !check_which_request(query, EVM_SIGN_TXN_REQUEST_TXN_DATA_TAG)) {
+    return status;
+  }
+  const evm_sign_txn_data_t *txn_data = &query->sign_txn.txn_data;
+  if (!txn_data->has_chunk_payload) {
+    evm_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                   ERROR_DATA_FLOW_INVALID_DATA);
+    return status;
+  }
+
+  // store the received transaction
+  if (NULL == txn_context->transaction) {
+    txn_context->transaction =
+        (uint8_t *)malloc(sizeof(common_chunk_payload_chunk_t));
+  }
+  const common_chunk_payload_chunk_t *chunk = &txn_data->chunk_payload.chunk;
+  txn_metadata dummy_metadata;
+  memcpy(txn_context->transaction, chunk->bytes, chunk->size);
+  // verify the received transaction
+  eth_byte_array_to_unsigned_txn(txn_context->transaction,
+                                 txn_context->transaction_size,
+                                 &txn_context->transaction_info,
+                                 &dummy_metadata);
+  eth_validate_unsigned_txn(&txn_context->transaction_info, &dummy_metadata);
```

Comment:
Return value left unchecked purposefully?

Reply:
not-intentional. updated